hibernate-core/src/main/java/org/hibernate/query/sqm/tree/expression/AbstractSqmExpression.java

```diff
@@ -125,7 +125,7 @@ public SqmPredicate in(Expression<?>... values) {
 
 	@Override
 	public SqmPredicate in(Collection<?> values) {
-		return nodeBuilder().in( this, values );
+		return nodeBuilder().in( this, (Collection) values );
 	}
```

Comment:
This is inspiring and error-prone. Without the above seemingly ugly fix, the following statement:
```
@Override
public SqmPredicate in(Collection<?> values) {
	return nodeBuilder().in( this, values );
}
```
would be routed to:
```
@Override
<T> SqmInPredicate<T> in(Expression<? extends T> expression, T... values);
```
not the intended alternative which accepts collection parameter:
```
@Override
<T> SqmInPredicate<T> in(Expression<? extends T> expression, Collection<T> values);
```
You can confirm this by following the link in IDE by mouse clicking. The end result is disastrous and the root cause is our imperfect usage of Java generics.